ground/gcs/src/plugins/config/configradiowidget.cpp

```diff
@@ -61,6 +70,61 @@ void ConfigRadioWidget::enableControls(bool enable)
     Q_UNUSED(enable);
 }
 
+void ConfigRadioWidget::roleChanged(bool ignored)
+{
+    (void) ignored;
+
+    if (ui->rb_tx->isChecked()) {
+        ui->groupTxSettings->setEnabled(true);
+        ui->groupRxSettings->setEnabled(false);
+    } else if (ui->rb_rx->isChecked()) {
+        ui->groupTxSettings->setEnabled(false);
+        ui->groupRxSettings->setEnabled(true);
+    } else {
+        ui->groupTxSettings->setEnabled(false);
+        ui->groupRxSettings->setEnabled(false);
+    }
+}
+
+void ConfigRadioWidget::refreshWidgetsValues(UAVObject *obj)
+{
+    (void) obj;
+
+    OpenLRS *olrsObj = OpenLRS::GetInstance(getObjectManager());
+
+    OpenLRS::DataFields openlrs = olrsObj->getData();
+
+    switch (openlrs.role) {
+    case OpenLRS::ROLE_DISABLED:
+    default:
```

Comment:
Shouldn't be too hard for ConfigTaskWidget to do radio buttons (if it doesn't already)... out of scope for this PR if you like.